### PR TITLE
Ea feature test

### LINKED ISSUE #12: Write feature test
---
name: feature test
about: article show page is rendering three youtube videos
---

**Is your feature request related to a problem? Please describe.** 

No.

**Describe the solution you'd like**

As a user,
On the article show page, 
I should see three youtube videos related to the tags of the article, 
in an aesthetically pleasing format.

** Additional context**

We now have a working mvp in development, so we need to build out the proper tests. This issue is to build out the feature test to confirm the videos are actually rendering on the article show page.

### PR DESCRIPTION
**Pull request by eamouzou**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Added test for youtube video descriptions on article show page
Added description of youtube video that includes search tags
## Added tests?

- [X] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/views/articles/_youtube.html.erb [hunk 1]
```diff
@@ -1,4 +1,10 @@
-<div class="youtube"></div>
+<div class="youtube-container">
+  <h3>Related Videos</h3>
+  <p>The videos below are based on the following tags: <%= @article.cached_tag_list %>.</p>
```
Comment (eamouzou) [inline app/views/articles/_youtube.html.erb:3]: @dreardon1021 Just added a short description; feel free to revise and change the test.

File: spec/features/articles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      visit "/#{user.username}"
+
+      expect(page).to have_content(article_1.title)
+      expect(page).to have_content(article_2.title)
+      expect(page).to have_content(last_article.title)
+      
+      click_on(last_article.title)
+
+      expect(page).to have_content('Related Videos')
+      expect(page).to have_content("The videos below are based on the following tags: #{tags}.")
```
Comment (Will-D-M) [inline spec/features/articles_spec.rb:23]: @eamouzou I like that we are confirming the video section is present, but is there any way to confirm the actual videos are present? I know Ian said it wasn't possible in Rspec, so if we can't get there by Monday, no big deal.

File: spec/views/articles_spec.rb [hunk 3]
```diff
@@ -1,50 +0,0 @@
...
-    expect(rendered).to have_text(Nokogiri::HTML(article1.processed_html).text)
-    expect(rendered).to have_css "div.body"
-  end
-
-  it "shows user new comment box" do
-    render
-    expect(rendered).to have_css("form#new_comment")
-    expect(rendered).to have_css("input#submit-button")
-  end
-end
```
Comment (Will-D-M) [inline spec/views/articles_spec.rb]: @eamouzou Is there a reason this spec was deleted? Is was imported with the origin dev.to file and think it is pretty in depth in covering the article show page. It covers tags and content, which I don't think are covered in the new spec you created.

**Comment by Will-D-M** [comment]
@eamouzou Left a few comments for your review.

File: app/views/articles/_youtube.html.erb (hunk 1 above)
Comment (Will-D-M) [reply app/views/articles/_youtube.html.erb:3]: @dreardon1021 Can you take a look at this and make sure the styling is aesthetically pleasing?

File: spec/features/articles_spec.rb (hunk 2 above)
Comment (eamouzou) [reply spec/features/articles_spec.rb:23]: @Will-Meighan, we can't really test for a fetch call with rspec. The video data is tested for in the api request test so I figure we've tested as fully as we can.

File: spec/views/articles_spec.rb (hunk 3 above)
Comment (eamouzou) [reply spec/views/articles_spec.rb]: @Will-Meighan good catch, that was not supposed to be deleted. I copied it to check something. I can restore it.

File: spec/features/articles_spec.rb (hunk 2 above)
Comment (Will-D-M) [reply spec/features/articles_spec.rb:23]: @eamouzou Okay coo, just wanted to make sure. This is good enough for me!

**Comment by Will-D-M** [comment]
Approved after review.

**Pull request merged**